src/graph_compute/io.cpp

```diff
@@ -94,3 +95,73 @@ int readEdgesFromFile(const string filepath,
 
   return edgelistfile_read(filepath, &builder);
 }
+
+void makeSimpleAndUndirected(vertex_t * nodes,
```

Comment:
I don't understand what this function is trying to achieve. Why is it changing the graph to not look like what the input files specified?